src/core/common/message.cpp

```diff
@@ -148,7 +148,7 @@ ThreadError MessagePool::FreeBuffers(Buffer *aBuffer)
 
 ThreadError MessagePool::ReclaimBuffers(int aNumBuffers)
 {
-    int numFreeBuffers;
+    uint16_t numFreeBuffers;
```

Comment:
Yea, I just tested it and it looks like your right..... I honestly didn't even know C would do that....